### ISSUE TITLE
[V2] "Choose icon" option doesn't work, and kills browsers

### ISSUE THREAD

**Issue by drpfenderson** [causefx/Organizr#1009]
###### Operating System: Gentoo (server)/Windows (browser)

<hr>

##### Problem Description:
When I go to the **Add New Tab** to add a new tab, the **Choose Icon** selector does not seem to pull up any icons, and instead starts eating my browser's memory for fun. Everything slows down, no icons come up, browser becomes unresponsive and  interface locks. Eventually browser says that page is slowing my browser down, and offers to kill it. This happens in both Chrome and Firefox on a recent, powerful PC - in both Normal and Incognito.

This is what I get when I kill the script in FF during a hang (my actual URL has been replaced with MYWEBSITEADDRESS, but no other alterations):
```
merge@http://MYWEBSITEADDRESS/js/jquery-2.2.4.min.js:2:3237
ca@http://MYWEBSITEADDRESS/js/jquery-2.2.4.min.js:3:3807
ua@http://MYWEBSITEADDRESS/js/jquery-2.2.4.min.js:3:12873
append@http://MYWEBSITEADDRESS/js/jquery-2.2.4.min.js:3:14539
Lang.prototype._mutation@http://MYWEBSITEADDRESS/js/jquery-lang.js:682:22
Lang.prototype.init/$.fn.append@http://MYWEBSITEADDRESS/js/jquery-lang.js:90:20
b</</c.prototype.template@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:13530
b</</c.prototype.option@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:9868
b</</c.prototype.append@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:7943
b</</c.prototype.bind/<@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:10049
b</</d.prototype.invoke@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:5209
b</</d.prototype.trigger@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:5028
b</</e.prototype.trigger@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:62779
b</</e.prototype._registerEvents/</<@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:61234
b</</d.prototype.query@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:34207
b</</e.prototype._registerEvents/<@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:61197
b</</d.prototype.invoke@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:5209
b</</d.prototype.trigger@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:5028
b</</e.prototype.trigger@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:62779
b</</e.prototype.open@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:62950
b</</e.prototype.toggleDropdown@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:62893
b</</e.prototype._registerSelectionEvents/<@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:60310
b</</d.prototype.invoke@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:5209
b</</d.prototype.trigger@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:5028
b</</e.prototype.bind/<@http://MYWEBSITEADDRESS/plugins/bower_components/custom-select/custom-select.min.js:1:16931
dispatch@http://MYWEBSITEADDRESS/js/jquery-2.2.4.min.js:3:7485
add/r.handle@http://MYWEBSITEADDRESS/js/jquery-2.2.4.min.js:3:5603
```

Otherwise, there are seemingly no other console errors.

**Note**: This function worked fine for me in v1 stable with the exact same browser setup.

<hr>

##### Reproduction Steps:
1 - Open page
2 - Navigate to Settings > Tab Editor
3 - Click the icon to add a new tab
4 - Watch as everything hangs

<hr>

#### Errors on screen?  If so paste here:

No errors on screen
```
```
##### Screenshot

![image](https://user-images.githubusercontent.com/7515881/50189585-b556c700-02da-11e9-82be-baacb73aaba6.png)

**Comment by causefx** [comment]
Here is my current exp on Chrome on windows:
https://ss.organizr.app/2018-12-18_16-43-47.gif

What browser are you using?

**Comment by drpfenderson** [comment]
Chrome: Version 71.0.3578.98 (Official Build) (64-bit) (Windows 10)
Firefox 64.0 (64-Bit) (Windows 10)

Just updated to most recent version of Chrome, opened Incognito, same deal. Everything slows to a crawl. It's weird. It's only on the "Add new tab" page. Browser starts to slow, but you can still put in new tab info. But trying to open the icon list is impossible. No other part of Organizr does this.

While it tried to load the list, CPU went to more than 40% and 3GB memory for ONLY that page. Can see by drilling down the Task Manager list.

Super weird. Very possibly me, but I have no clue what cause this on just that page. As soon as I "disable script" on browser's recommendation, everything goes back to normal.

**Comment by causefx** [comment]
so changing the icon on the edit tab is fine?

**Comment by drpfenderson** [comment]
I'm sorry - that's my fault for lack of clarity. I meant the Tab
Organization page is where things start to slow down. It will sometimes
hang when it brings up the edit/new tab as well... But not all the time.
However, 100% of the time, it hangs when I try to add an icon. Adding a
pic, like the basic ones that come with Organize in the Photo section,
works just fine. It's the icon search dropdown - the one that looks so
silly smooth in your GIF above.

**Comment by christronyxyocum** [comment]
I've seen this before when PHP-FPM tuning has not been done. I would suggest that you follow [THIS GUIDE](https://technicalramblings.com/blog/optimizing-php-fpm-to-get-faster-load-times-of-tabs-in-organizr/) and see if performance increases.

**Comment by Cookie-Monster-Coder** [comment]
Any update on this problem? I'm having the same issue.

**Comment by causefx** [comment]
"fixed" on ed5895d
